sdk/search/Azure.Search.Documents/tests/Samples/FieldBuilder/SerializePropertyNamesAsCamelCaseAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Azure.Search.Documents.Tests.Samples
+{
+    /// <summary>
+    /// Indicates that the public properties of a model type should be serialized as camel-case in order to match
+    /// the field names of a search index.
+    /// </summary>
+    /// <remarks>
+    /// Types without this attribute are expected to have property names that exactly match their corresponding
+    /// fields names in Azure Cognitive Search. Otherwise, it would not be possible to use instances of the type to populate
+    /// the index.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct, AllowMultiple = false, Inherited = true)]
+    public class SerializePropertyNamesAsCamelCaseAttribute : Attribute
```

Comment:
I don't think we should bring this one forward.  Whatever we do with ObjectSerializer should handle this for us.

Reply:
This is a straight-up port of track 1 and the final API will look much different.